src/wallet/test/wallet_tests.cpp

```diff
@@ -334,6 +334,82 @@ BOOST_AUTO_TEST_CASE(LoadReceiveRequests)
     BOOST_CHECK_EQUAL(values[1], "val_rr1");
 }
 
+// Test some watch-only wallet methods by the procedure of loading (LoadWatchOnly),
+// checking (HaveWatchOnly), getting (GetWatchPubKey) and removing (RemoveWatchOnly) a
+// given PubKey, resp. its corresponding P2PK Script. Results of the the impact on
+// the address -> PubKey map is dependent on whether the PubKey is a point on the curve
+static void TestWatchOnlyPubKey(CWallet& wallet, const CPubKey& add_pubkey)
+{
+    CScript p2pk = GetScriptForRawPubKey(add_pubkey);
+    CKeyID add_address = add_pubkey.GetID();
+    CPubKey found_pubkey;
+    LOCK(wallet.cs_wallet);
+
+    // all Scripts (i.e. also all PubKeys) are added to the general watch-only set
+    BOOST_CHECK(!wallet.HaveWatchOnly(p2pk));
+    wallet.LoadWatchOnly(p2pk);
+    BOOST_CHECK(wallet.HaveWatchOnly(p2pk));
+
+    // only PubKeys on the curve shall be added to the watch-only address -> PubKey map
+    bool isPubKeyFullyValid = add_pubkey.IsFullyValid();
+    if (isPubKeyFullyValid) {
+        BOOST_CHECK(wallet.GetWatchPubKey(add_address, found_pubkey));
+        BOOST_CHECK(found_pubkey == add_pubkey);
+    } else {
+        BOOST_CHECK(!wallet.GetWatchPubKey(add_address, found_pubkey));
+        BOOST_CHECK(found_pubkey == CPubKey());
+    }
+
+    wallet.RemoveWatchOnly(p2pk);
+    BOOST_CHECK(!wallet.HaveWatchOnly(p2pk));
+
+    if (isPubKeyFullyValid)
```

Comment:
nit: braces

As above, you should also check that `found_pubkey` is the default `CPubKey`.

Reply:
Since `GetWatchPubKey` doesn't modify the passed PubKey if it returns `false`, it would still have the same contents as before, so in this case the correct assertion would be `found_pubkey == add_pubkey`. To not confuse the reader (one could wrongly assume that this assert checks for an explicit modification by the function in this case as well), I add the comment `// passed key is unchanged` for those two cases.